### PR TITLE
feat: activity types

### PR DESCRIPTION
**Pull request by AceLikesGhosts**
Adds support for [Custom Activity Types](https://discord.com/developers/docs/topics/gateway-events#activity-object-activity-types)

### CODE REVIEW

**Comment by AceLikesGhosts** [comment]
😐Just realised some of the styling was changed in different files, didn't intend for that. Will revert.

File: plugins/customrpc/components/RPCSettings.tsx [hunk 1]
```diff
@@ -12,8 +12,17 @@ const {
   Category,
   ErrorBoundary,
   SwitchItem,
+  RadioItem
 } = components;
 
+const ActivityTypes = {
```
Comment (Tharki-God) [inline plugins/customrpc/components/RPCSettings.tsx:18]: would be better as enum ig

Comment (AceLikesGhosts) [reply plugins/customrpc/components/RPCSettings.tsx:18]: Enums make it fucky-wucky as the professionals would say due to it being a two way mapped object. Current method works and doesnt result in.. this.
image

**Comment by E-boi** [comment]
It looks like "url" is never set in the activity. [Take a look how I set a rpc](https://github.com/E-boi/replugged-plugins/blob/main/plugins/customrpc/index.ts#L111-L150) and set the "url" then should be good to merge.

**Comment by AceLikesGhosts** [comment]
> It looks like "url" is never set in the activity. [Take a look how I set a rpc](https://github.com/E-boi/replugged-plugins/blob/main/plugins/customrpc/index.ts#L111-L150) and set the "url" then should be good to merge.

Fixed (:

**Comment by AceLikesGhosts** [comment]
yay

**Pull request merged**